examples/never-type/never_type.bal

```diff
@@ -0,0 +1,44 @@
+import ballerina/io;
+import ballerina/lang.'xml as xmllib;
+
+type Person record {
+    string name;
+    int age;
+};
+
+type Student record {
+    string name;
+    // You can specify an optional field with `never` type, but can not assign a value.
+    never gender?;
+};
+
+// This function never returns. So we have return type of `never`.
+function validateAndPanic(int age) returns never {
+    if (age < 18) {
+        error e = error("Invalid Age");
+        panic e;
+    }
+}
```

Comment:
Is it correct to say that this function never returns? If age >= 18, this function completes execution normally. Doesn't that mean it returns? @hasithaa

Reply:
I also don't think this is correct. Not returning and not panicking is the same as returning nil.

I think we should just have a function that panics either way.

And maybe **not** call it from the main function?

Reply:
I'll remove the if condition and have only the `panic error` part. That will work.